dangerfile.js

```diff
@@ -1,27 +1,33 @@
 // eslint-disable-next-line import/no-extraneous-dependencies, no-unused-vars, object-curly-newline
-import { danger, warn, fail, message } from 'danger';
+import { danger, warn, fail } from 'danger';
 
-const newChangelog = danger.git.created_files.filter((filePath) => {
-  const srcFilePattern = /^packages\/terra-([a-z-])*\/CHANGELOG.md/i;
-  return srcFilePattern.test(filePath);
-});
+const determinePackages = (changedList, path) => {
+  const path1 = path.split('packages/')[1];
+  const packageName = path1.split('/')[0];
+  return changedList.includes(packageName) ? changedList : changedList.concat([packageName]);
+};
 
-const modifiedChangelog = danger.git.modified_files.filter((filePath) => {
-  const srcFilePattern = /^packages\/terra-([a-z-])*\/CHANGELOG.md/i;
-  return srcFilePattern.test(filePath);
-});
+const newChangelogs = danger.git.created_files.filter((filePath) => {
+  const changelogPattern = /^packages\/terra-([a-z-])*\/CHANGELOG.md/i;
+  return changelogPattern.test(filePath);
+}).reduce(determinePackages, []);
 
-const modifiedSrcFiles = danger.git.modified_files.filter((filePath) => {
-  const srcFilePattern = /^packages\/terra-([a-z-])*\/src/i;
-  return srcFilePattern.test(filePath);
-});
+const modifiedChangelogs = danger.git.modified_files.filter((filePath) => {
+  const changelogPattern = /^packages\/terra-([a-z-])*\/CHANGELOG.md/i;
+  return changelogPattern.test(filePath);
+}).reduce(determinePackages, []);
 
-const hasCHANGELOGChanges = modifiedChangelog.length > 0 || newChangelog.length > 0;
-const hasModifiedSrcFiles = modifiedSrcFiles.length > 0;
+const modifiedPackages = danger.git.modified_files.filter((filePath) => {
+  const changelogPattern = /^packages\/terra-([a-z-])*\/CHANGELOG.md/i;
+  const packagePattern = /^packages\/terra-([a-z-])*\//i;
+  return packagePattern.test(filePath) && !changelogPattern.test(filePath);
+}).reduce(determinePackages, []);
 
-// Fail if there are src code changes without a CHANGELOG update
-if (hasModifiedSrcFiles && !hasCHANGELOGChanges) {
-  fail('Please include a CHANGELOG entry with this PR.');
+const missingChangelogs = modifiedPackages.filter(packageName => !modifiedChangelogs.includes(packageName) || newChangelogs.includes(packageName));
+
+// Fail if there are package changes without a CHANGELOG update
+if (missingChangelogs.length > 0) {
+  fail(`Please include a CHANGELOG entry for each changed package this PR. Looks like a CHANGELOG is missing for: \n - ${missingChangelogs.join('\n - ')}`);
 }
 
 // Warn when there is a big PR
```

Comment:
Lets get rid of the big pr warning. I don't think it's valuable.

Reply:
Nathan is making this change in : #2418